packaging/debian/storagenode.config

```diff
@@ -144,41 +146,65 @@ case "$STATE" in
             else
                   go_back
             fi
-      ;;
+	    ;;
       "$IP_ADDRESS")
-            db_input high storagenode/external_addr || true
-            if db_go; then
-                  db_get storagenode/external_addr
-                  ip_port="$RET"
-                  ip="${ip_port%:*}"
-                  port="${ip_port##*:}"
-
-                  if ! valid_ip $ip ;  then
-                        error_ask_continue storagenode/external_addr/not_ip_port_continue
-                  elif ! valid_port $port ;  then
-                        error_ask_continue storagenode/external_addr/not_ip_port_continue
-                  fi
-                  next_state
-            else
-                  go_back
-            fi
-      esac
+          db_input high storagenode/external_addr || true
+          if db_go; then
+              db_get storagenode/external_addr
+              ip_port="$RET"
+              ip="${ip_port%:*}"
+              port="${ip_port##*:}"
+
+              if ! valid_ip $ip ;  then
+                  error_ask_continue storagenode/external_addr/not_ip_port_continue
+              elif ! valid_port $port ;  then
+                  error_ask_continue storagenode/external_addr/not_ip_port_continue
+              fi
+              next_state
+          else
+              go_back
+          fi
+	  ;;
+      "$STORAGEDIR")
+          db_input high storagenode/storagedir || true
+          if db_go; then
+              db_get storagenode/storagedir
+              storagedir="$RET"
+
+	      if [ ! -d "$storagedir" ]; then
+		  error_ask_continue storagenode/storagedir/does_not_exist
+              fi
+              next_state
+          else
+              go_back
+          fi
+	  ;;
+      "$DISKSPACE")
+          db_input high storagenode/diskspace || true
+          if db_go; then
+              db_get storagenode/diskspace
+              diskspace="$RET"
+	      if [[ ! "$diskspace" =~ ^[0-9]+[GTgt]{1} ]]; then
+		  error_ask_continue storagenode/diskspace/wrong_format
+	      else
+		  space_amount="${diskspace%*[GTgt]}"
```

Comment:
I think this will work, but the `*` has no effect here so it's a little confusing. Can we take it out?